src/Sylius/Bundle/UserBundle/Authentication/AuthenticationFailureHandler.php

```diff
@@ -13,18 +13,44 @@
 
 namespace Sylius\Bundle\UserBundle\Authentication;
 
+use Psr\Log\LoggerInterface;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
 use Symfony\Component\Security\Core\Exception\AuthenticationException;
 use Symfony\Component\Security\Http\Authentication\DefaultAuthenticationFailureHandler;
+use Symfony\Component\Security\Http\HttpUtils;
+use Symfony\Contracts\Translation\TranslatorInterface;
 
 final class AuthenticationFailureHandler extends DefaultAuthenticationFailureHandler
 {
+    /**
+     * @param list<mixed> $options
+     */
+    public function __construct(
+        HttpKernelInterface $httpKernel,
+        HttpUtils $httpUtils,
+        array $options = [],
+        ?LoggerInterface $logger = null,
+        private readonly ?TranslatorInterface $translator = null,
+    ) {
+        parent::__construct($httpKernel, $httpUtils, $options, $logger);
+    }
+
     public function onAuthenticationFailure(Request $request, AuthenticationException $exception): Response
     {
         if ($request->isXmlHttpRequest()) {
-            return new JsonResponse(['success' => false, 'message' => $exception->getMessageKey()], 401);
+            return new JsonResponse([
+                'success' => false,
+                'message' => $this->translator?->trans(
+                    $exception->getMessageKey(),
+                    [],
+                    'security',
+                    $request->getLocale(),
+                ) ?? $exception->getMessageKey(),
+                Response::HTTP_UNAUTHORIZED,
```

Comment:
The response code is the second parameter of the `JsonResponse` object:
```suggestion
            ], Response::HTTP_UNAUTHORIZED);
```